Fixing meta data tags env and environment

kwoodson opened the pull request.

detiber commented on playbooks/adhoc/grow_docker_vg/grow_docker_vg.yml:
```diff
@@ -151,7 +151,7 @@
       tags:
         Name: "{{ ec2_tag_Name }}"
-        env: "{{ ec2_tag_environment }}"
+        env: "{{ oo_clusterid }}"
```
shouldn't this be clusterid: "{{ oo_clusterid }}" instead?

detiber commented on playbooks/aws/openshift-cluster/tasks/launch_instances.yml:
```diff
@@ -2,8 +2,8 @@
-    env: "{{ cluster }}"
-    env_host_type: "{{ cluster }}-openshift-{{ type }}"
+    cluster: "{{ cluster }}"
+    cluster_host_type: "{{ cluster }}-openshift-{{ type }}"
```
do we need cluster_host_type still? shouldn't tooling query for the separate tags?

detiber commented:
@kwoodson is this just to address the renaming of env to cluster_id? Will the re-addition of env be handled in a separate PR?

kwoodson commented:
@detiber,

I have added the environment back in.  Updated some of the clusterid references.

detiber commented on playbooks/adhoc/grow_docker_vg/grow_docker_vg.yml:
```diff
@@ -151,7 +151,7 @@
       tags:
         Name: "{{ ec2_tag_Name }}"
-        env: "{{ ec2_tag_environment }}"
+        env: "{{ ec2_tag_clusterid }}"
```
shouldn't this be `'clusterid: "{{ ec2_tag_clusterid }}"` ?

detiber commented on bin/cluster:
```diff
@@ -290,6 +291,8 @@ if __name__ == '__main__':
                                help='number of infra nodes to create in cluster')
     create_parser.add_argument('-e', '--etcd', default=0, type=int,
                                help='number of external etcd hosts to create in cluster')
+    create_parser.add_argument('--env', default='prod', type=str,
```
Should this default to prod? or should we have it default to something more dev related?

kwoodson replied:
@detiber,

Most of our customers will be classified as 'prod' environment.  I don't think devs care much about the environment since we aren't currently using it.  That said, I'm up for suggestions.

detiber replied:
@kwoodson agreed, but bin/cluster is used beyond our internal usage. Prod just seems like an odd default for tooling meant to operate on more than one 'environment'

kwoodson replied:
I understand that its beyond our internal usage.  int/stg/prod are all suggestions here.  Could use dev?  test?  Any other suggestions?

detiber replied:
either dev or test seems appropriate to me.

kwoodson commented:
@detiber, set the environment default to 'dev'

detiber commented:
@kwoodson :+1: from me on the changes. I suspect the other playbooks (gce, libvirt, openstack) will also need to be updated for these changes though.

kwoodson commented:
@lhuard1A, @menren, @twiest 

Could you please weigh in on these changes?

menren commented:
Any file containing `tag_env`, `tag_host-type`,  `tag_env-host_type`,  `tag_environment` are potentially impacted.


On AWS some files have to be modified
For instance on GCE, the following files are impacted and need to be modified analogously to AWS : 
- openshift-cluster/tasks/launch_instances.yml
- openshift-cluster/service.yml
- openshift-cluster/wip.yml
- openshift-cluster/update.yml
- openshift-cluster/config.yml
- openshift-cluster/join_node.yml
- openshift-cluster/terminate.yml
- openshift-cluster/list.yml

kwoodson commented:
@detiber, @lhuard1A, @menren, @twiest, @abutcher, 

I have removed tag_env, added back tag_environment, and have updated the original env to be clusterid.  env_host-type has also been removed.  I have tested this on aws and on gce.

Please test on byo, libvirt, openstack, etc and leave any feedback.

menren commented:
Hello, I will test it ASAP for GCE

menren commented:
With this patch for termination playbook, it's good for GCE : 
```
diff --git a/playbooks/gce/openshift-cluster/terminate.yml b/playbooks/gce/openshift-cluster/terminate.yml
index 9e527f3..faa46c0 100644
--- a/playbooks/gce/openshift-cluster/terminate.yml
+++ b/playbooks/gce/openshift-cluster/terminate.yml
@@ -12,7 +12,7 @@
       groups: oo_hosts_to_terminate
       ansible_ssh_user: "{{ deployment_vars[deployment_type].ssh_user | default(ansible_ssh_user, true) }}"
       ansible_sudo: "{{ deployment_vars[deployment_type].sudo }}"
-    with_items: (groups['tag_environment-' ~ cluster_env]|default([])) | groups['tag_clusterid-' ~ cluster_id] | default([], true) | difference(['localhost']) | difference(groups.status_terminated | default([], true))
+    with_items: (groups['tag_clusterid-' ~ cluster_id] | default([])) | difference(['localhost'])

```

kwoodson commented:
@menren, I have applied your patch.

menren commented:
@kwoodson :+1: for GCE

kwoodson commented:
@lhuard1A, @detiber, @twiest, @abutcher,

Would you look over these changes?  I have tested on aws and gce.  I'll wait until after the new year to merge this but I would like to know that its good to go.

detiber commented:
I'm testing out a patch set that would clean this up a bit, will submit a PR once I verify it works.
